File: jenkins-common/src/main/java/com/app/jenkins_common/lib/StringPreferenceType.java
package com.app.jenkins_common.lib;


import android.content.SharedPreferences;


public final class StringPreferenceType implements SharedPreferenceType<String> {
  private final SharedPreferences sharedPreferences;
  private String defaultValue = "";

  public StringPreferenceType(final SharedPreferences sharedPreferences) {
    this.sharedPreferences = sharedPreferences;
  }

  @Override
  public boolean notNullPref() {
    return sharedPreferences != null;
  }

  @Override
  public synchronized String get(final String key) {
    if (notNullPref()) {
      return sharedPreferences.getString(key, defaultValue);
    } else {
      return defaultValue;
    }
  }

  @Override
  public boolean isSet(final String key) {
    return sharedPreferences.contains(key);
  }

  @Override
  public synchronized void set(final String key, final String value) {
    if (notNullPref()) {
      final SharedPreferences.Editor editor = sharedPreferences.edit();
      editor.putString(key, value);
      editor.apply();
    }
  }

  @Override
  public synchronized void delete(final String key) {
    if (notNullPref()) {
      sharedPreferences.edit().remove(key).apply();
    }
  }
}

File: jenkins-common/src/main/java/com/app/jenkins_common/lib/StringPreferenceType.kt
package com.app.jenkins_common.lib


import android.content.SharedPreferences


class StringPreferenceType(private val sharedPreferences: SharedPreferences?) : SharedPreferenceType<String> {
    private val defaultValue = ""

    override fun notNullPref(): Boolean {
        return sharedPreferences != null
    }

    override fun get(key: String): String {
        return if (notNullPref()) {
            sharedPreferences!!.getString(key, defaultValue)
        } else {
            defaultValue
        }
    }

    override fun isSet(key: String): Boolean {
        return sharedPreferences!!.contains(key)
    }

    @Synchronized override fun set(key: String, value: String) {
        if (notNullPref()) {
            val editor = sharedPreferences!!.edit()
            editor.putString(key, value)
            editor.apply()
        }
    }

    @Synchronized override fun delete(key: String) {
        if (notNullPref()) {
            sharedPreferences!!.edit().remove(key).commit()
        }
    }
}
